### PR TITLE
Partially extracted spring, partially supported micronaut

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: _build-and-test-all.sh [hunk 1]
```diff
@@ -30,8 +30,9 @@ if [[ "${DATABASE}" == "mysql" ]] && [[ "${BROKER}" == "kafka" ]]; then
     ./wait-for-services.sh $DOCKER_HOST_IP "actuator/health" 8099
 
     export TEST_CUSTOM_DB=true
-    ./gradlew cleanTest :eventuate-tram-db-broker-integration-test:test -Dtest.single=TramIntegrationCustomDBTest
-    ./gradlew cleanTest :eventuate-tram-commands-db-broker-integration-test:test -Dtest.single=TramCommandsDBBrokerIntegrationCustomDBTest
+
+    ./gradlew cleanTest :eventuate-tram-db-broker-integration-test:test --tests "io.eventuate.tram.broker.db.integrationtests.TramIntegrationCustomDBTest"
+    ./gradlew cleanTest :eventuate-tram-commands-db-broker-integration-test:test --tests "io.eventuate.tram.commands.db.borker.integrationtests.TramCommandsDBBrokerIntegrationCustomDBTest"
```
Comment (cer) [inline _build-and-test-all.sh]: `borker` is mispelt

Comment (dartartem) [reply _build-and-test-all.sh]: I will fix

File: eventuate-tram-producer-jdbc-micronaut/src/main/java/io/eventuate/tram/messaging/producer/jdbc/micronaut/TramMessageProducerJdbcFactory.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import io.eventuate.tram.messaging.producer.jdbc.MessageProducerJdbcImpl;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.annotation.Value;
+
+import javax.inject.Singleton;
+import javax.sql.DataSource;
+
+@Factory
+//@Requires(property = "micronaut.eventuate.tram.message.producer.jdbc.factory", value = "true")
```
Comment (cer) [inline eventuate-tram-producer-jdbc-micronaut/src/main/java/io/eventuate/tram/messaging/producer/jdbc/micronaut/TramMessageProducerJdbcFactory.java]: Please remove

File: eventuate-tram-producer-jdbc/src/main/java/io/eventuate/tram/messaging/producer/jdbc/MessageProducerJdbcImpl.java [hunk 3]
```diff
@@ -5,21 +5,25 @@
 import io.eventuate.common.jdbc.EventuateSchema;
 import io.eventuate.tram.messaging.common.Message;
 import io.eventuate.tram.messaging.producer.common.MessageProducerImplementation;
-import org.springframework.beans.factory.annotation.Autowired;
+
+import java.sql.SQLException;
```
Comment (cer) [inline eventuate-tram-producer-jdbc/src/main/java/io/eventuate/tram/messaging/producer/jdbc/MessageProducerJdbcImpl.java]: SQLException should not appear anywhere!

File: eventuate-tram-producer-jdbc/src/main/java/io/eventuate/tram/messaging/producer/jdbc/MessageProducerJdbcImpl.java [hunk 4]
```diff
@@ -32,11 +36,16 @@ public String generateMessageId() {
...
-            message.getHeaders(),
-            eventuateSchema);
+    try {
+      eventuateCommonJdbcOperations.insertIntoMessageTable(message.getId(),
+              message.getPayload(),
+              message.getRequiredHeader(Message.DESTINATION),
+              currentTimeInMillisecondsSql,
+              message.getHeaders(),
+              eventuateSchema);
+    } catch (SQLException e) {
```
Comment (cer) [inline eventuate-tram-producer-jdbc/src/main/java/io/eventuate/tram/messaging/producer/jdbc/MessageProducerJdbcImpl.java]: Most definitely do not do this.

File: eventuate-tram-producer-jdbc-micronaut/src/main/java/io/eventuate/tram/messaging/producer/jdbc/micronaut/TramMessageProducerJdbcFactory.java (hunk 2 above)
Comment (dartartem) [reply eventuate-tram-producer-jdbc-micronaut/src/main/java/io/eventuate/tram/messaging/producer/jdbc/micronaut/TramMessageProducerJdbcFactory.java]: ok

File: eventuate-tram-producer-jdbc/src/main/java/io/eventuate/tram/messaging/producer/jdbc/MessageProducerJdbcImpl.java (hunk 3 above)
Comment (dartartem) [reply eventuate-tram-producer-jdbc/src/main/java/io/eventuate/tram/messaging/producer/jdbc/MessageProducerJdbcImpl.java]: I got it

File: eventuate-tram-producer-jdbc/src/main/java/io/eventuate/tram/messaging/producer/jdbc/MessageProducerJdbcImpl.java (hunk 4 above)
Comment (dartartem) [reply eventuate-tram-producer-jdbc/src/main/java/io/eventuate/tram/messaging/producer/jdbc/MessageProducerJdbcImpl.java]: ok, it was a temporary 'solution'

**Comment by cer** [comment]
Merged into branch wip-micronaut

**Pull request closed without merging**